src/toucan/db.clj

```diff
@@ -152,6 +152,21 @@
   [& body]
   `(do-in-transaction (fn [] ~@body)))
 
+(defn do-in-transaction-with-rollback
+  "Execute F inside a DB transaction with rollback (will not commit changes on complete). Prefer macro form
+  `transaction-with-rollback` to using this directly."
+  [f]
+  (jdbc/with-db-transaction [conn (connection)]
+    (jdbc/db-set-rollback-only! conn)
+    (binding [*transaction-connection* conn]
+    (f))))
+
+(defmacro transaction-with-rollback
```

Comment:
@camsaul My idea here was that something like this can be useful on tests that modify the database (the test can run wrapped in a transaction that would no be commited, thus not affecting other tests).

I though doing could be ok. There could also be some other options:
* Modify the defmacro transaction to accept a parameter that then decides to rollback
* Make this new macro inside the `util.test` namespace (because I believe this would almost exclusively be used on tests?)